tests/functions_tests/test_reshape.py

```diff
@@ -22,6 +22,7 @@ def check_forward(self, x_data):
         shape = self.gy.shape
         x = chainer.Variable(x_data)
         y = functions.reshape(x, shape)
+        self.assertEqual(y.data.dtype, numpy.float32)
```

Comment:
`dtype` of inputs of `reshape` can be int32. So we must check the case where x has dtype of `int32` (that will be another issue). This testcase itself is OK because input `x` is assumed to be `float32`.